File: 2856 Minimum Array Length After Pair Removals/2856minimum-array-length-after-pair-removals.py
class Solution:
    def minLengthAfterRemovals(self, nums: List[int]) -> int:
        n, i = len(nums), 0
        j = (n + 1) // 2
        p = 0

        
        while i < n // 2 and j < n:
            if nums[i] != nums[j]:
                i += 1
                j += 1
                p += 1
            else:
                j += 1
            
        return n - 2*p

File: 2856 Minimum Array Length After Pair Removals/2856minimum-array-length-after-pair-removals.cpp
class Solution {
public:
    int minLengthAfterRemovals(vector<int>& nums) {
        int n = nums.size(), i = 0, p = 0;
        int j = n / 2;

        while (i < n / 2 and j < n){
            if (nums[i] != nums[j]){
                i++;
                j++;
                p++;
            } else {
                j++;
            }
        }
        return n - 2*p;
    }
};